d2ir/d2ir.go

```diff
@@ -663,29 +679,73 @@ func (m *Map) EnsureField(kp *d2ast.KeyPath, refctx *RefContext) (*Field, error)
 		}
 		i++
 	}
-	return m.ensureField(i, kp, refctx)
+
+	var fa []*Field
+	err := m.ensureField(i, kp, refctx, create, &fa)
+	return fa, err
 }
 
-func (m *Map) ensureField(i int, kp *d2ast.KeyPath, refctx *RefContext) (*Field, error) {
+func (m *Map) ensureField(i int, kp *d2ast.KeyPath, refctx *RefContext, create bool, fa *[]*Field) error {
+	us, ok := kp.Path[i].Unbox().(*d2ast.UnquotedString)
+	if ok && us.Pattern != nil {
+		fa2, ok := m.doubleGlob(us.Pattern)
+		if ok {
+			if i == len(kp.Path)-1 {
+				*fa = append(*fa, fa2...)
+			} else {
+				for _, f := range fa2 {
+					if f.Map() == nil {
+						f.Composite = &Map{
+							parent: f,
+						}
+					}
+					err := f.Map().ensureField(i+1, kp, refctx, create, fa)
+					if err != nil {
+						return err
+					}
+				}
+			}
+			return nil
+		}
+		for _, f := range m.Fields {
```

Comment:
you can put this in the else of the above right

Reply:
I don't see how. This is the code for single globs, the above is the code for double globs. They have different behaviour.